checker/src/main/java/org/checkerframework/checker/lock/LockStore.java

```diff
@@ -226,17 +226,12 @@ boolean hasLockPossiblyHeld(CFValue value) {
     }
 
     @Override
-    public void insertValue(JavaExpression je, @Nullable CFValue value) {
-        if (value == null) {
-            // No need to insert a null abstract value because it represents
-            // top and top is also the default value.
-            return;
-        }
+    public void insertValueImpl(JavaExpression je, @Nullable CFValue value) {
```

Comment:
It's weird that the "helper" method is the one that is overridden.

Reply:
Do you have a different suggestion?  I don't think this pattern is so uncommon.  Its purpose here is to avoid boilerplate that would otherwise appear at the beginning of every method.  I think it would be uglier and more error-prone to expect every overriding method to include that boilerplate.

Reply:
I would move the boilerplate code into a method, `shouldInsert`, that subclasses can call.  `shouldInsert` should also not be `final` in case subclasses wish to change the behavior.  (For example, a subclasses may wish to insert some nondeterministic expression, but not others.)